atom/browser/api/atom_api_app.cc

```diff
@@ -807,6 +807,14 @@ void App::BrowserChildProcessKilled(
 void App::RenderProcessReady(content::RenderProcessHost* host) {
   ChildProcessLaunched(content::PROCESS_TYPE_RENDERER,
                        host->GetProcess().Handle());
+
+  // TODO(jeremy): this isn't really the right place to be creating
+  // `WebContents` instances, but this was implicitly happening before in
+  // `RenderProcessPreferences`, so this is at least more explicit...
+  content::WebContents* web_contents =
+      AtomBrowserClient::Get()->GetWebContentsFromProcessID(host->GetID());
+  if (web_contents)
+    WebContents::FromOrCreate(v8::Isolate::GetCurrent(), web_contents);
```

Comment:
IIUC the previous version in `IsWebContents` was only there to make sure we update preferences for `BrowserWindow` or `WebView` and not for a devtools `Remote` type. Since the ipc was sent when render process being created without knowing what its contents were. But in this PR, the ipc is made by the renderer process so it chooses the right target `WebContents`, so i don't think this wrapper creation is necessary.

Reply:
If this line is deleted, devtools stops opening.

Reply:
@zcbenz whats the intent behind wrapping `api::WebContents` at render process start for devtools ?

Reply:
Hmm I am not sure why this is related to devtools, I would need to do some debugging to know.

Reply:
Given that this preserves the existing behaviour (i.e. calling `WebContents::FromOrCreate` on every render process at creation time), I'm going to go ahead and merge this, but it does seem odd.